src/client/auth/sasl.rs

```diff
@@ -0,0 +1,98 @@
+use crate::{
+    bson::{doc, spec::BinarySubtype, Binary, Bson, Document},
+    bson_util,
+    client::auth::AuthMechanism,
+    error::{Error, Result},
+};
+
+/// Encapsulates the command building of a `saslStart` command.
```

Comment:
Since there's a bit of duplicate functionality between the auth mechanisms in terms of building the `saslStart` and `saslContinue` commands and reading the server responses, I pulled it out from `scram` into a separate module. These are essentially mini-operations, although I didn't use the `Operation` trait since they won't ever be sent through `execute_operation`, making a lot of the `Operation` functionality overkill for them.